### DISCUSSION TITLE
Send HR to VRC Chatbox

### DISCUSSION THREAD

**Discussion by SouljaVR** [200Tigersbloxed/HRtoVRChat_OSC#11] [Ideas]
As the title says, would be cool if heartrate could be sent to the new VRC chatbox. And if possible still sent to the avatar too. It supports a new message every 1.5s, would be perfect for this :)

**Comment by 200Tigersbloxed** [comment]
> would be cool if heartrate could be sent to the new VRC chatbox

Is there any documentation for this?

**Reply by SouljaVR** [reply]
yes, here you go:

https://ask.vrchat.com/t/developer-update-11-august-2022/12286

"Also, we support OSC input! The OSC endpoints are:

    /chatbox/input s b Input text into the chatbox. If b is True, send the text in s immediately, bypassing the keyboard. If b is False, open the keyboard and populate it with the provided text.
    /chatbox/typing b Toggle the typing indicator on or off.

Right now the chatbox’s OSC input endpoint only supports ASCII characters. We want to be able to support UTF-8 via OSC, so we’re going to look into solutions for that.

The chatbox itself supports UTF right now, so the aforementioned issue is only with OSC."

Killfrenzy TTS already added support and its opensource: https://github.com/killfrenzy96/KillFrenzyAvatarText/

**Reply by SouljaVR** [reply]
Got confused, i meant TTS by VRCWizard: https://github.com/VRCWizard/TTS-Voice-Wizard/releases

**Reply by 200Tigersbloxed** [reply]
> Any update im trying to do this from scratch but it's driving me insane, ive seen it done before so can someone write up a tutorial on how to display in the chatbox vs on the avatar itself.

It's currently planned to be implemented. You can track progress on issue #13 (https://github.com/200Tigersbloxed/HRtoVRChat_OSC/issues/13)
